### DISCUSSION TITLE
Quickfix/J Performance in JMeter - Unlocking trading platform performance monitoring via FIX

### DISCUSSION THREAD

**Discussion by vinipx** [quickfix-j/quickfixj#672] [Q&A]
Hi folks, 

Is there anyone here who happened to **simulate/test the performance** results of trading platforms/systems and quickfix/j engines using Apache JMeter (e.g. TCP Sampler)? 

I'm looking for examples or ideas on how to onboard QuickFix/J message execution via **Apache JMeter** in order to collect metrics and monitor a trading platform I'm currently working on. 

_Note: I currently have functional tests written in Java, but JMeter would be more preferable for specific perf testing and due to project and people preference/expertise)_

Any ideas are welcome!

Thank you very much

**Comment by chrjohn** [comment]
Disclaimer: I've never used JMeter, only had a quick look at their documentation.
Can't you implement your own class which does some message sending using https://jmeter.apache.org/api/org/apache/jmeter/protocol/java/sampler/JavaSamplerClient.html

This https://stackoverflow.com/a/70108230/4962355 suggests you could also use Groovy Scripts to send messages if you don't want to implement your own JavaSamplerClient.

You probably can also use TCP Sampler but need to construct the FIX messages on your own then and send them as text. If you want to go this route, you probably are better off disabling some checks in QuickFIX/J to make your life easier, otherwise you need to pre-calculate the checksum and send unique sequence numbers on each message.

```
ValidateSequenceNumbers=N 
ValidateChecksum=N
```
